### PR TITLE
Update release process docs

### PR DESCRIPTION
**Pull request by amyblais**
- Cleaned up release docs to remove any outdated checklists since we are using Playbooks to run checklists.
 - Updated release docs to reflect that we are now in a bi-weekly release cadence for Cloud releases (instead of a weekly cadence).

### CODE REVIEW

File: operations/research-and-development/product/release-process/release-overview.md
Comment (lindalumitchell) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
Mattermost ships with a new version on the 16th of each month in [binary form](https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html#upgrading-to-the-latest-version) for self-managed customers. Mattermost also ships releases with new features and bug fixes to [Mattermost Cloud](https://docs.mattermost.com/about/cloud-subscriptions.html) biweekly.
```

File: operations/research-and-development/product/release-process/cloud-release.md
Comment (lindalumitchell) [inline operations/research-and-development/product/release-process/cloud-release.md]:
```suggestion
    - Each QA signs off as appropriate, using the [QA Approval playbook](https://community.mattermost.com/playbooks/playbooks/qeay15ou6frsmnu6tpmup9mxbc) run for that release.
```

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (lindalumitchell) [inline operations/research-and-development/product/release-process/desktop-release.md]:
```suggestion
    - [QA approval](https://community.mattermost.com/playbooks/playbooks/h798dt39mpbymb8z5uoiuf4hdo/outline) should be ready by T-2.
```

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (lindalumitchell) [inline operations/research-and-development/product/release-process/mobile-release.md]:
```suggestion
    - [QA approval](https://community.mattermost.com/playbooks/playbooks/8dx9zrs16bbg8eg1hdhw35zdua) should be ready by T-2.
```

File: operations/research-and-development/product/release-process/feature-release.md
Comment (lindalumitchell) [inline operations/research-and-development/product/release-process/feature-release.md]:
```suggestion
    - [QA approval](https://community.mattermost.com/playbooks/playbooks/8dx9zrs16bbg8eg1hdhw35zdua) should be ready by T-2.
```

**Review by lindalumitchell — DISMISSED** [review]
Looks great, thanks Amy!

File: operations/research-and-development/product/release-process/cloud-release.md
Comment (saturninoabril) [inline operations/research-and-development/product/release-process/cloud-release.md]: Not by 5pm but it's the latest, right?
```suggestion
    - QA approval should be given latest by 5pm Eastern on Monday so that the release rollout can be started early Tuesday morning. 
```

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (saturninoabril) [inline operations/research-and-development/product/release-process/desktop-release.md:13]: Not related to this line but for the line above regarding the quarterly ship.

> Desktop App releases follow a fixed schedule of 4 releases per year. Releases ship quarterly by the 16th of March, June, September, and December of each year.

I wonder if 16th of Dec is a good timing to do the release given during this period (before and after) most people are taking holiday breaks and the support maybe limited. Thoughts @m1lt0n @amyblais?

File: operations/research-and-development/product/release-process/playbooks-release.md
Comment (saturninoabril) [inline operations/research-and-development/product/release-process/playbooks-release.md:5]: Playbooks has recently moved to `mattermost-server`, built-in as product. I wonder if plugin release is still accurate or has to be updated. Same case about the [Boards release doc](https://github.com/mattermost/mattermost-handbook/blob/amyblais-patch-2/operations/research-and-development/product/release-process/boards-release.md). Maybe something for @sbishel to confirm?

File: operations/research-and-development/product/release-process/release-overview.md
Comment (saturninoabril) [inline operations/research-and-development/product/release-process/release-overview.md]: Same here in pre-packaging Playbooks and Boards if still accurate given they are to be shipped as built-in products.

File: operations/research-and-development/product/release-process/release-overview.md
Comment (saturninoabril) [inline operations/research-and-development/product/release-process/release-overview.md]:
```suggestion
    - QA approval should be given latest by 5pm Eastern on Monday so that the release rollout can be started early Tuesday morning. 
```

**Review by saturninoabril — COMMENTED** [review]
Thanks @amyblais for the quick updates! Overall looks good to me, except for few clarifications.

**Review by m1lt0n — DISMISSED** [review]

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (enahum) [inline operations/research-and-development/product/release-process/mobile-release.md]: I would add here to cut an RC based on the release branch and start testing. here is the reason

1. the weekly beta builds are always based of main branch and can be used for progressing testing until we branch for release, once branched, beta could have code that is not going to be part of the release.
2. Give more time to QA to run their test suite (automated and manual)
3. Identify problems with the release early

Thoughts @saturninoabril

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (enahum) [inline operations/research-and-development/product/release-process/mobile-release.md]: here I would add cutting another RC based of the release branch

**Review by enahum — DISMISSED** [review]
I'm going to approve as my suggestions will require some changes with the QA team that should be discussed first.

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (amyblais) [reply operations/research-and-development/product/release-process/desktop-release.md:13]: I think releasing in December is fine, but I'll let @m1lt0n and @devinbinnie comment if they have additional thoughts.

File: operations/research-and-development/product/release-process/playbooks-release.md
Comment (amyblais) [reply operations/research-and-development/product/release-process/playbooks-release.md:5]: Monorepo didn't make it to v7.10 release, so I'll let Scott to confirm how this works now.

File: operations/research-and-development/product/release-process/release-overview.md
Comment (amyblais) [reply operations/research-and-development/product/release-process/release-overview.md]: Monorepo didn't make it to v7.10 release, so I'll let @sbishel to confirm how this works now.

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (amyblais) [reply operations/research-and-development/product/release-process/mobile-release.md]: I agree with this suggestion, I'll also add @larkox to share any additional thoughts.

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (m1lt0n) [reply operations/research-and-development/product/release-process/desktop-release.md:13]: Initially that schedule was created to help spread the QA reviews of the internal QA people. Since the situation has change, we're open to a different schedule (we'd prefer to stick to the fixed cadence, but if different months e.g. Mid-April, July, October and January make more sense now, we're open to it)

@devinbinnie Any preference or adjustments we want to make here?

File: operations/research-and-development/product/release-process/playbooks-release.md
Comment (sbishel) [reply operations/research-and-development/product/release-process/playbooks-release.md:5]: Other than security patches, neither boards nor playbooks will cut plugins any more.

File: operations/research-and-development/product/release-process/release-overview.md
Comment (sbishel) [reply operations/research-and-development/product/release-process/release-overview.md]: Other than security patches, neither boards nor playbooks will cut plugins any more.

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (saturninoabril) [reply operations/research-and-development/product/release-process/desktop-release.md:13]: Thanks @m1lt0n! Sounds good to me. @lindalumitchell thoughts?

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (saturninoabril) [reply operations/research-and-development/product/release-process/mobile-release.md]: Thanks @enahum and I totally agree. I honestly thought that we normally do progressive testing during release period for mobile app. I didn't realize that "the weekly beta builds are always based of main branch" is the case.

**Review by m1lt0n — DISMISSED** [review]

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (larkox) [inline operations/research-and-development/product/release-process/mobile-release.md]: Just for my knowledge. What is the difference between being in feature complete, and having the release branch cut? My main concern is that if no features get into the code after T-28, but the release branch is not cut until T-14, we would be "blocking" the merge of features those 14 days... every month. Shouldn't feature complete be at the same time as we cut the release branch? (I may be misunderstanding or mixing concepts, so please bear with me)

Comment (larkox) [inline operations/research-and-development/product/release-process/mobile-release.md]: Once the release branch is created, how often do we want to create builds (release candidates of sort) for QA? We can keep creating betas from main, but as stated before, once the release branch is created, the beta (out of main) may contain code that is not meant to release. So on those 14 days since the release branch is created, at least 1 beta will be released out of sync with the release.

In the meanwhile, some fixes may be cherry picked to the release branch.

We could create a new build after every cherry pick, or a more forgiving cadence (once a day if there has been any merge, or even once a week). 0/5 on the final details, but I feel we should at least discuss it.

**Review by larkox — DISMISSED** [review]
With the changes proposed by Elias, LGTM. Nevertheless, there are two minor discussion point open, but I don't want to block this PR based on those.

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (amyblais) [reply operations/research-and-development/product/release-process/mobile-release.md]: Makes sense, I've edited it so that feature complete is at T-14.

Comment (amyblais) [reply operations/research-and-development/product/release-process/mobile-release.md]: Also adding @lindalumitchell for any thoughts.

Comment (amyblais) [reply operations/research-and-development/product/release-process/mobile-release.md]: Adding @lindalumitchell & @saturninoabril for any thoughts.

**Review by lindy65 — DISMISSED** [review]
Thanks @amyblais - made a couple of spelling/grammar updates. Otherwise looks good to me 👍

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (lindalumitchell) [reply operations/research-and-development/product/release-process/desktop-release.md:13]: Thanks for asking! Given that people being out in mid- and late December might crunch testing time for a January release (possibly even more than a December release), I'd vote for either leaving it as-is, or shifting it in the other direction, e.g. Feb, May, Aug, Nov. But I'm fine either way.

**Review by lindalumitchell — DISMISSED** [review]
Thanks, LGTM!

File: operations/research-and-development/product/release-process/mobile-release.md
Comment (lindalumitchell) [reply operations/research-and-development/product/release-process/mobile-release.md]: Thanks, I also agree, and apparently had confusion as well. :)

Comment (lindalumitchell) [reply operations/research-and-development/product/release-process/mobile-release.md]: Thanks, I think at most one new build per day for testing (if there have been changes / cherry picks of course) would be often enough.

Comment (saturninoabril) [reply operations/research-and-development/product/release-process/mobile-release.md]: > Once the release branch is created, how often do we want to create builds (release candidates of sort) for QA?

+1 to what Linda said.

@larkox How do we know if the build is from main or release branch during the release testing period or in general? How about alpha for main and beta build for release branch?

Comment (larkox) [reply operations/research-and-development/product/release-process/mobile-release.md]: I will leave this to @enahum , since changing the name of the app to support this, may not be possible (or practical). We could potentially add a new parameter to the build to use on the build info (build name?) along the build number, build version, etc... But not sure that is the best idea neither.

Comment (enahum) [reply operations/research-and-development/product/release-process/mobile-release.md]: Oh boy, this is getting out of hand and will create a lot of confusion, it already started with community members.

Cutting builds have an impact, specially on the stores so having a cadence of daily builds is not something that I will encourage.

Having a 3rd app is also not an option, 2 should be more than enough as it allows us to have the latest and gratest as well as what is actually being shipped.

The idea is to cover our based but without over complicating things.

As I mentioned in the QA channel, the idea is to coordinate between the teams to make minimize the pain, not to add more :) 

That said, we can build as many as we want like PR's that do not increase the build numbers nor are submitted to the stores (including testflight)

Comment (saturninoabril) [reply operations/research-and-development/product/release-process/mobile-release.md]: I understand and agree to limit into 2 apps (beta and release), and definitely agree about team coordination especially during release testing. Also noted about the daily build. It may not likely to happen during the early phase of release testing but may be requested as we get closer to the deadline in a case where a critical bug fix is necessary to ship.

Currently, when beta builds are posted at `Release: Mobile Apps`, there's an info about `Version` and `Build Number` ([example](https://community.mattermost.com/core/pl/m7uhfq1g1tyabrj958686cgjxc)). How about adding extra fields such as `Branch` and `Commit (short sha)` so that we, in QA, can easily identify the base branch where it's built?

Comment (amyblais) [reply operations/research-and-development/product/release-process/mobile-release.md]: I've made edits to [the Mobile release process section](https://github.com/mattermost/mattermost-handbook/pull/1192/files#diff-2589aadc650609878898e858a1cd2ce69dd31bba3c001f94be7182d8f797a031) based on the discussions here.

**Review by enahum — APPROVED** [review]

**Review by larkox — APPROVED** [review]

**Review by saturninoabril — APPROVED** [review]
Thanks @amyblais, LGTM!

File: operations/research-and-development/product/release-process/desktop-release.md
Comment (saturninoabril) [reply operations/research-and-development/product/release-process/desktop-release.md:13]: Seems like all are fine as is, and so I'm good with it as well.

**Pull request merged**
